### PR TITLE
DIP-6 Adaptation

### PR DESCRIPTION
**Pull request by JayJay1024**

### CODE REVIEW

File: src/hooks/use-collator-power.ts [hunk 1]
```diff
@@ -101,34 +77,31 @@ export const useCollatorPower = (
...
                     }
                     return acc;
                   },
-                  { stakedDeposit: 0n, stakedRing: 0n, stakedKton: 0n }
+                  { stakedDeposit: 0n, stakedRing: 0n }
                 );
-                const power = stakingToPower(stakedRing + stakedDeposit, stakedKton, ringPool, ktonPool);
 
-                const commission = collatorCommission[cur] || "0.00%";
-                return { ...acc, [cur]: commissionWeightedPower(power, commission) };
```
Comment (hackfisher) [inline src/hooks/use-collator-power.ts:131]: @AurevoirXavier `commissionWeightedPower` still take effects.

But I will patch this version first

**Comment by hackfisher** [comment]
Test looks so far so good, so tag latest commit of this PR for a production patch:

https://github.com/darwinia-network/collator-staking-ui/releases/tag/v0.12.6-patch

File: src/hooks/use-collator-power.ts (hunk 1 above)
Comment (JayJay1024) [reply src/hooks/use-collator-power.ts:131]: Oh, it should be that for Darwinia that has not yet been upgraded, commission needs to be considered, right?

Comment (hackfisher) [reply src/hooks/use-collator-power.ts:131]: I think commission still works for Crab.

**Comment by hackfisher** [comment]
image

> members who did not unbond kton during the transition period are now not able to see them in the staking dashboard to unbond.  Two people have seen the same problem.  Addresses are:
> 0x9Fa44Ea6483d70e99C6b3c402596eDf56fD7882c

File: src/hooks/use-collator-power.ts (hunk 1 above)
Comment (JayJay1024) [reply src/hooks/use-collator-power.ts:131]: > I think commission still works for Crab.

Is that so? @AurevoirXavier

File: src/components/unbond-kton-modal.tsx [hunk 2]
```diff
@@ -62,6 +62,7 @@ export default function UnbondKtonModal({
         symbol={ktonToken.symbol}
         decimals={ktonToken.decimals}
         balance={stakedKton}
+        max={0n}
```
Comment (hackfisher) [inline src/components/unbond-kton-modal.tsx]: @JayJay1024

File: src/hooks/use-collator-power.ts (hunk 1 above)
Comment (hackfisher) [reply src/hooks/use-collator-power.ts:131]: https://github.com/darwinia-network/darwinia/blob/351a0b34377e236dcee2fef6620c020c69bc27d4/pallet/staking/src/lib.rs#L744-L767

Comment (JayJay1024) [reply src/hooks/use-collator-power.ts:131]: It has been changed so that active-collator displays `Vote`, and waiting-collator displays `Total-staked`

**Comment by hackfisher** [comment]
image
Good but the votes for Darwinia is all zero now, could we do some adapt here?

https://impersonator.xyz/?address=0x9Fa44Ea6483d70e99C6b3c402596eDf56fD7882c&url=https://staking-ag5tki3la-itering.vercel.app

File: src/hooks/use-collator-power.ts (hunk 1 above)
Comment (acgxv) [reply src/hooks/use-collator-power.ts:131]: It should be `commissionWeightedVote`. Exclude `Power`, but retain everything else.

**Review by hackfisher — APPROVED** [review]

**Pull request merged**